src/mixins/people-roles-mixin.ts

```diff
@@ -197,7 +205,18 @@ export default class PeopleRolesMixin extends Vue {
     this.resetData()
   }
 
-  protected onRemovePerson (index: number): void {
+  protected async onRemovePerson (index: number): Promise<void> {
+    const orgPerson = this.orgPersonList[index]
+    const isProprietor = this.isProprietor(orgPerson)
+
+    if (isProprietor && this.isPerson(orgPerson)) {
+      if (!await this.confirmRemoveProprietorPerson()) return
+    } else if (isProprietor && this.isOrganization(orgPerson)) {
+      if (!await this.confirmRemoveProprietorOrganization()) return
+    } else {
+      if (!await this.confirmRemove()) return
+    }
```

Comment:
In addition to consistency, an advantage of moving the confirmation out of ListPeopleAndRoles and into here is that the default confirmation is prompted if you click the Delete button while editing a person. (Previously there was no confirmation in this case.)